doc/extensions/extension-sdk-developer-guide.md

```
@@ -57,86 +51,437 @@ type MyPolicy struct {
+
+Add kubebuilder RBAC markers to your reconciler. At minimum, every extension needs:
+
+```go
+//+kubebuilder:rbac:groups=extensions.kuadrant.io,resources=mypolicies,verbs=get;list;watch;update;patch
+//+kubebuilder:rbac:groups=extensions.kuadrant.io,resources=mypolicies/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=extensions.kuadrant.io,resources=mypolicies/finalizers,verbs=update
+```
+
+Add additional markers if your reconciler interacts with other resources:
+
+```go
```

markdownlint-cli2 (0.23.2)

[warning] Code block style
Expected: indented; Actual: fenced

(MD046, code-block-style)